lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -823,15 +846,54 @@ def upload_reads(self, ctx, params):
         self.log('Starting upload reads, parsing args')
         o, wsid, name, objid, kbtype, single_end, fwdid, revid, shock = (
             self._proc_upload_reads_params(params))
-        interleaved = 1 if (not single_end and not revid) else 0
+        #IF from shock fwdid and revid are shock nodes, if not shock they are file paths
+        dfu = DataFileUtil(self.callback_url)
+        fwdpath = None
+        revpath = None
```

Comment:
No need for these two lines, fwd and revpath are always set in the if/else below